src/Utils/Responsive/Responsive.tsx

```diff
@@ -71,7 +71,8 @@ export function createResponsiveComponents<M extends string>() {
       isSupportedEnvironment = () => {
         return (
           typeof window !== "undefined" &&
-          typeof window.matchMedia !== "undefined"
+          typeof window.matchMedia !== "undefined" &&
+          window.name !== "nodejs"
```

Comment:
Because we use jsdom, these tests were passing, which in turn would lead to the media queries being kicked off and thus ignoring the initial matched one passed in as a prop.

Reply:
@alloy - this is how we deal with it in our tests, which I think is simple and no nonsense:  

```
https://github.com/damassi/reaction/blob/51146296989d3a13c9d9409506074f25b8810df7/src/Styleguide/Components/__tests__/ArtistCard.test.tsx#L27-L29
```

Reply:
I don’t think that’s a good solution as it is very surprising having to do this. Is there any reason to consider the test env a support env?

Reply:
Can you elaborate a bit? Functions on `window` points to `global.window` which can be stubbed to whatever, per common jest conventions. If `matchMedia` is null, then the responsive component dimension fires immediately, as it it were on the server, and thus we can test different dimensions. This seems simpler than adding environment specific config into the component itself, which is leaky.

Reply:
Oh I’m fine with moving that example you pointed to to `setup_jest.ts` so it applies to all tests, I just don’t want anybody having to spend any more time on having to figure this issue out.

(What I was asking about is if there’s ever a reason to not shortcut, regardless of implementation.)

Reply:
So what's the consensus here? Emptying out `matchMedia` in `setup_jest.ts` or using a check in the `isSupportedEnvironment` method? Looking through the issues, it seems like the jsdom team recommends using `navigator.userAgent.includes("jsdom")` to test if it's in jsdom.

Reply:
I’ll update it to empty it in `setup_jest.ts`